### PR TITLE
#150324119 Delete Bookmarked Contributions and Paths

### PR DESCRIPTION
**Pull request by dennomwas**
### What does this PR do?

Adds a feature to delete a bookmarked contribution or path.

### Description of Tasks completed
- Create a delete contribution endpoint
- add delete path bookmark logic
- add delete contribution logic
- Write associated tests

#### How should this be manually tested? 

After cloning the repo cd into it and run;
```
pip install -r requirements.txt
```
Url for the endpoint
```
http://127.0.0.1:8000/api/v1/bookmarks/<string:id>
```
#### To run the tests
```
pytest
```
#### What are the relevant pivotal tracker stories? ####
 #150324119

### CODE REVIEW

**Comment by andela-uibrahim** [comment]
this branch has 14 commits and 16 file changes. I think that is too much for the feature

**Comment by andela-oowonikoko** [comment]
@dennomwas Please rebase this branch against develop :thank_you:

File: api/views/bookmarks.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        # query db to get the bookmark id
+        else:
+            bookmark = Bookmark.query.get(bookmark_id)
+            # check if the bookmark is a path
+            if bookmark:
+                path_bookmark = bookmark.query.filter_by(
+                    path_id=Bookmark.path_id).first()
+                # delete the path bookmark
+                if path_bookmark:
+                    db.session.delete(bookmark)
```
Comment (andela-ojoloko) [inline api/views/bookmarks.py]: I belive all references to `db.session.delete` and `db.session.commit` can be replaced with `bookmark.delete()`

Comment (dennomwas) [reply api/views/bookmarks.py]: @andela-ojoloko I am doing another PR with a new branch, this logic has been refactored...thanks

**Pull request closed without merging**